homeassistant/components/thermostat/nest.py

```diff
@@ -109,6 +120,24 @@ def turn_away_mode_off(self):
         """ Turns away off. """
         self.structure.away = False
 
+    @property
+    def min_temp(self):
+        """ Identifies min_temp in Nest API or defaults if not available. """
+        temp = self.device.away_temperature.low
+        if temp is None:
+            return super().min_temp
+        else:
+            return round(self.hass.config.temperature(temp, TEMP_CELCIUS)[0])
+
+    @property
+    def max_temp(self):
+        """ Identifies mxn_temp in Nest API or defaults if not available. """
+        temp = self.device.away_temperature.high
+        if temp is None:
+            return super().max_temp
```

Comment:
I think it should be that a thermostat should not be aware of the user `unit_of_measurement` preference. Instead it should always return all values in its own `unit_of_measurement`. It is the responsibility of the class `Thermostat` to convert it when it is exposing it to the outside world.

This means that the conversion from thermostat to state has to happen [over here](https://github.com/balloob/home-assistant/blob/master/homeassistant/components/thermostat/__init__.py#L136-L137).

We can then convert the `min_temp` default value in the `Thermostat` class to be:
```python
    @property
    def min_temp(self):
        """ Return minimum temperature. """
        return convert(7, TEMP_CELCIUS, self.unit_of_measurement)
```

To make sure that every `Thermostat` object has a `unit_of_measurement`, we should probably add it to the `Thermostat` class:
```python
    @property
    def unit_of_measurement(self):
        """ Unit of measurement this thermostat expresses itself in. """
        raise NotImplementedError
```